redash/serializers/__init__.py

```diff
@@ -245,7 +245,7 @@ def serialize_dashboard(obj, with_widgets=False, user=None, with_favorite_state=
 
     d = {
         "id": obj.id,
-        "slug": obj.slug,
+        "slug": obj.name_as_slug,
```

Comment:
If we want to update the "slug" from the backend we also need to update other Dashboard resource urls to use the Id instead of the slug, e.g.: the Archive, favorite and unfavorite ones.